client/src/store/restaurantSelectors.js

```diff
@@ -0,0 +1,32 @@
+// @flow
+
+import { createSelector } from 'reselect';
+import type { AppState, Restaurant } from '../flowTypes';
+
+const restaurantSelector = (state: AppState) => state.restaurants.restaurants;
+const cuisineSelector = (state: AppState) => state.restaurants.filters.cuisine;
+const ratingSelector = (state: AppState) => state.restaurants.filters.rating;
+const maxDeliveryTimeSelector = (state: AppState) => state.restaurants.filters.maxDeliveryTime;
+const nameSelector = (state: AppState) => state.restaurants.filters.name;
+
+export const filteredRestaurants = createSelector(
+  [restaurantSelector, cuisineSelector, ratingSelector, maxDeliveryTimeSelector, nameSelector],
+  (restaurants, cuisine, rating, maxDeliveryTime, name) => {
+    return restaurants.filter((restaurant: Restaurant): boolean => {
+      let isMatched = true;
```

Comment:
It's pretty same as `restaurant.name.includes(name) && restaurant.cuisine === cuisine && ...`, right? 🤔

Reply:
Not necessarily, because I want to pass the filter check if the filter value is falsy (=if the user didn't use the filter). However it could be cleaner, so I just made a small refactor for this :)